### PR TITLE
Feature/38 file refresh

### PR DESCRIPTION
**Pull request by SzymonWelter**

### CODE REVIEW

File: Orchestrator/Prognetics.Saga.Orchestrator/SagaOrchestrator.cs [hunk 1]
```diff
@@ -1,33 +1,50 @@
...
+        _logger = logger;
     }
 
-    public void Start(ISagaSubscriber sagaSubscriber)
+    public async Task Start(CancellationToken cancellationToken = default)
     {
-        _sagaSubscriber = sagaSubscriber;
-        IsStarted = true;
+        await _sagaClient.Initialize();
+        _sagaSubscriber = await _sagaClient.GetSubscriber();
```
Comment (SzymonWelter) [inline Orchestrator/Prognetics.Saga.Orchestrator/SagaOrchestrator.cs:28]: pass cancellation token

File: Orchestrator/Prognetics.Saga.Orchestrator/SagaOrchestrator.cs [hunk 2]
```diff
@@ -1,33 +1,50 @@
...
+        _sagaClient = sagaClient;
+        _logger = logger;
     }
 
-    public void Start(ISagaSubscriber sagaSubscriber)
+    public async Task Start(CancellationToken cancellationToken = default)
     {
-        _sagaSubscriber = sagaSubscriber;
-        IsStarted = true;
+        await _sagaClient.Initialize();
```
Comment (SzymonWelter) [inline Orchestrator/Prognetics.Saga.Orchestrator/SagaOrchestrator.cs:27]: pass cancellation token

File: Orchestrator/Prognetics.Saga.Orchestrator/SagaHost.cs [hunk 3]
```diff
@@ -1,46 +1,45 @@
...
-    {
-        if (disposing) {
-            _client.Dispose();
+            using var scope = _serviceScopeProvider.CreateAsyncScope();
+            using var orchestrator = scope.ServiceProvider.GetRequiredService<IStartableSagaOrchestrator>();
+            await orchestrator.Start(restartToken);
+
+            while (!restartToken.IsCancellationRequested)
+            {
+                await Task.Delay(
```
Comment (SzymonWelter) [inline Orchestrator/Prognetics.Saga.Orchestrator/SagaHost.cs]: Task.FromCancelled

File: Parsers/Prognetics.Saga.Parser.Json/Reader/JsonFromInternetTransactionLedgerReader.cs [hunk 4]
```diff
@@ -1,24 +1,73 @@
...
+                    callback(transactionLedger);
+                    _versionHash = currentFileHash;
+                }
+            }
+            catch(Exception e)
+            {
+                onError(e);
+            }
+
+            await Task.Delay(_readerConfiguration.TrackingInterval, cancellation);
```
Comment (SzymonWelter) [inline Parsers/Prognetics.Saga.Parser.Json/Reader/JsonFromInternetTransactionLedgerReader.cs:70]: Task.FromCancellation?

**Review by ChrisGawlinski — APPROVED** [review]

**Pull request merged**